integration_tests/ipp/test.sh

```diff
@@ -5,32 +5,91 @@ MODULE_DIR=$(dirname $0)
 ZGRAB_ROOT=$MODULE_DIR/../..
 ZGRAB_OUTPUT=$ZGRAB_ROOT/zgrab-output
 
+OUTPUT_ROOT=$ZGRAB_OUTPUT/ipp
+
 mkdir -p $ZGRAB_OUTPUT/ipp
 
-CONTAINER_NAME=zgrab_ipp
+versions="cups cups-tls"
+
+function test_cups() {
+    echo "ipp/test: Tests runner for ipp_cups"
+
+    CONTAINER_NAME="zgrab_ipp_cups" $ZGRAB_ROOT/docker-runner/docker-run.sh ipp --timeout 3 --verbose > "$OUTPUT_ROOT/cups.json"
+    # FIXME: No good reason to use a tmp file & saved file, b/c I'm not testing any failure states yet
+    #CONTAINER_NAME="zgrab_ipp_cups" $ZGRAB_ROOT/docker-runner/docker-run.sh ipp --timeout 3 --verbose > out.tmp
+    major=$($ZGRAB_ROOT/jp -u data.ipp.result.version_major < "$OUTPUT_ROOT/cups.json")
+    minor=$($ZGRAB_ROOT/jp -u data.ipp.result.version_minor < "$OUTPUT_ROOT/cups.json")
+    response=$($ZGRAB_ROOT/jp -u data.ipp.result.response < "$OUTPUT_ROOT/cups.json")
+    cups=$($ZGRAB_ROOT/jp -u data.ipp.result.cups_version < "$OUTPUT_ROOT/cups.json")
+    rm -f out.tmp
+    if ! [ $major = "2" ]; then
+        echo "ipp/test: Incorrect major version. Expected 2, got $major"
+        exit 1
+    fi
+    if ! [ $minor = "1" ]; then
+        echo "ipp/test: Incorrect minor version. Expected 1, got $minor"
+        exit 1
+    fi
+    if ! [ $cups = "CUPS/2.1.3" ]; then
+        echo "ipp/test: Incorrect CUPS version. Expected CUPS/2.1.3, got $cups"
+    fi
+}
 
-OUTPUT_FILE=$ZGRAB_OUTPUT/ipp/ipp.json
+function test_cups_tls() {
+    echo "ipp/test: Tests runner for ipp_cups"
+
+    CONTAINER_NAME="zgrab_ipp_cups-tls" $ZGRAB_ROOT/docker-runner/docker-run.sh ipp --timeout 3 --ipps --verbose > "$OUTPUT_ROOT/cups-tls.json"
+    # FIXME: No good reason to use a tmp file & saved file, b/c I'm not testing any failure states yet
+    #CONTAINER_NAME="zgrab_ipp_cups-tls" $ZGRAB_ROOT/docker-runner/docker-run.sh ipp --timeout 3 --ipps --verbose > out.tmp
+    major=$($ZGRAB_ROOT/jp -u data.ipp.result.version_major < "$OUTPUT_ROOT/cups-tls.json")
+    minor=$($ZGRAB_ROOT/jp -u data.ipp.result.version_minor < "$OUTPUT_ROOT/cups-tls.json")
+    response=$($ZGRAB_ROOT/jp -u data.ipp.result.response < "$OUTPUT_ROOT/cups-tls.json")
+    cups=$($ZGRAB_ROOT/jp -u data.ipp.result.cups_version < "$OUTPUT_ROOT/cups-tls.json")
+    tls=$($ZGRAB_ROOT/jp -u data.ipp.result.tls < "$OUTPUT_ROOT/cups-tls.json")
```

Comment:
Here it might be safer to check for a field in e.g. `data.mysql.result.tls.handshake_log.server_certificates.certificate.parsed.subject_dn` -- you still might have some of the `tls` structure (e.g. the `client_hello`) even if the server doesn't give a valid response.